ui/src/utils/api-examples/index.js

```
@@ -0,0 +1,205 @@
+ *
+ * @param {string} language - Language key (e.g., 'go', 'python', 'curl')
+ * @param {string} serviceName - Service name (e.g., 'queue', 'auth')
+ * @param {string} apiVersion - API version (e.g., 'v1')
+ * @param {object} entry - API entry metadata from references.json
+ * @param {string} version - Taskcluster version (e.g., '93')
+ * @returns {string|null} Code example or null if not found
+ */
+function generateSingleExample(
+  language,
+  serviceName,
+  apiVersion,
+  entry,
+  version = DEFAULT_VERSION
+) {
+  // Generate payload example if entry has input schema
+  let payloadExample = null;
+
+  if (entry.input) {
+    const schemaId = `/schemas/${serviceName}/${entry.input}`;
+    const schema = getSchemaContent(schemaId);
+
+    if (schema) {
+      // Pass all references for $ref resolution
+      payloadExample = generatePayloadExample(schema, references);
+    }
+  }
+
+  const generators = {
+    curl: () =>
+      generateCurlExample(serviceName, apiVersion, entry, payloadExample),
+    go: () =>
+      generateGoExample(
+        serviceName,
+        apiVersion,
+        entry,
+        version,
+        payloadExample
+      ),
+    python: () =>
+      generatePythonExample(
+        serviceName,
+        apiVersion,
+        entry,
+        false,
+        payloadExample
+      ),
+    pythonAsync: () =>
+      generatePythonExample(
+        serviceName,
+        apiVersion,
+        entry,
+        true,
+        payloadExample
+      ),
+    node: () =>
+      generateNodeExample(serviceName, apiVersion, entry, payloadExample),
+    web: () =>
+      generateWebExample(serviceName, apiVersion, entry, payloadExample),
+    rust: () =>
+      generateRustExample(serviceName, apiVersion, entry, payloadExample),
+    shell: () =>
+      generateShellExample(serviceName, apiVersion, entry, payloadExample),
+  };
+  const generator = generators[language];
+
+  if (!generator) {
+    // Unknown language requested
```

I would probably prefer an assert here and early failure or even exception, if someone is using this function incorrectly to avoid silencing errors